tests/Node.py

```diff
@@ -1221,7 +1221,10 @@ def getNextCleanProductionCycle(self, trans):
         # The voted schedule should be promoted now, then need to wait for that to become irreversible
         votingTallyWindow=120  #could be up to 120 blocks before the votes were tallied
         promotedBlockNum=self.getHeadBlockNum()+votingTallyWindow
-        self.waitForIrreversibleBlock(promotedBlockNum, timeout=rounds/2)
+        # There was waitForIrreversibleBlock but due to bug it was waiting for head and not lib.
```

Comment:
Go ahead and fix `waitForIrreversibleBlock` which should be as simple as changing the default argument. It is fine to leave this as you have it, but should probably update this comment.

Reply:
I had fixed it, which is why this is taking longer. He could have extended the test, but since it had never failed before, we presumed it was fine to replace this call with what it had been doing before (unintentionally)

Reply:
I'v fine with it using `waitForBlock`, but we should fix `waitForIrreversibleBlock` which is still:
```
    def waitForIrreversibleBlock(self, blockNum, timeout=WaitSpec.default(), blockType=BlockType.head):
        return self.waitForBlock(blockNum, timeout=timeout, blockType=blockType)
```

Reply:
ahh, what I fixed was waitForBlock to actually honor blockType, I missed that waitForIrreversibleBlock looked like this.  Which then makes me wonder why Dmytro's change had any effect.

Reply:
@heifner perhaps you are looking at wrong branch.
```
    def waitForIrreversibleBlock(self, blockNum, timeout=WaitSpec.default()):
        return self.waitForBlock(blockNum, timeout=timeout, blockType=BlockType.lib)
```

Reply:
Apparently so.